packages/rich-text-editor/src/vaadin-rich-text-editor-mixin.js

```diff
@@ -315,6 +363,15 @@ export const RichTextEditorMixin = (superClass) =>
         });
       });
 
+      this._editor.on('editor-change', () => {
+        const selection = this._editor.getSelection();
+        if (selection) {
+          const format = this._editor.getFormat(selection.index, selection.length);
+          this.style.setProperty('--_color-value', format.color || null);
```

Comment:
Adding these private css variables to the `<vaadin-rich-text-editor>` host component as inline styles might be confusing:
![Screenshot 2024-05-13 at 11 18 35](https://github.com/vaadin/web-components/assets/1222264/1bd50fb4-892e-47c1-a693-18d1ddab941d)

We could consider adding them to one of the elements inside the shadow root, such as `.vaadin-rich-text-editor-container` or `[part="toolbar"]`

Reply:
Updated to set these custom properties on `[part="toolbar"]` as they are only used by toolbar buttons.